src/OpenSage.Game/Data/Ini/Parser/IniParser.cs

```diff
@@ -320,7 +320,19 @@ public int[] ParseInLineIntegerArray()
 
         public uint ParseUnsignedInteger() => ScanUnsignedInteger(GetNextToken());
 
-        private long ScanLong(IniToken token) => Convert.ToInt64(token.Text);
+        private long ScanLong(IniToken token)
+        {
+            try
+            {
+                return (long)Convert.ToUInt64(token.Text);
+            }
+            catch(Exception _)
+            {
+                if (token.Text.StartsWith("-"))
```

Comment:
What kind of input value causes this code to run?

Reply:
Sth like 999999999999999

Reply:
Pretty sure this should be `Convert.ToInt64(token.Text)`, otherwise you're throwing away every negative number. I haven't yet been able to find an example of an out-of-range number like `999999999999999` (and I have installed the Edain mode). Can you post a snippet of an .ini file that has a value like that?

Reply:
![image](https://user-images.githubusercontent.com/10086882/50546152-4b67e700-0c23-11e9-9f0d-fd0d6aabab39.png)

Reply:
true, ill change it to `Convert.ToInt64(token.Text)` then

Reply:
I think the method implementation should be:

``` csharp
private long ScanLong(IniToken token)
{
    try
    {
        return Convert.ToInt64(token.Text);
    }
    catch (OverflowException)
    {
        return token.Text.StartsWith("-")
            ? long.MinValue
            : long.MaxValue;
    }
}
```

Reply:
I'd suggest token.Text[0] == '-' as it's a lot faster and it's a given the text is more than 0 chars in length if an overflow occurred.

Reply:
Also are you sure you want to use Int64 instead of Int32? I can't think of a value from the games which would actually be 64 bits.

Reply:
thats true, so should we remove long completely??

Reply:
We need ParseLong for a few things, but it doesn’t need to be used for new fields in this PR. I’m still not sure we’ve got the behaviour right. I assume Generals’ ini parser used sscanf, which seems to “crop” out-of-range values to “infinity” rather than crashing. Maybe we should try to do that.

Reply:
Well since this stuff only occurs in the edain mod and not in any of the vanilla games, i would guess the modders just wanted some kind of max value. So our current implementation is sufficient for that, no need to overthink it, its just for durations and ranges, where infinity would be overkill.

Reply:
I agree. Please could you change it to the implementation I posted above, then I can merge this?

Reply:
done